joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/passes/PhpSetKnownTypes.scala

```diff
@@ -0,0 +1,109 @@
+package io.joern.php2cpg.passes
+
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.passes.CpgPass
+import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.PropertyNames
+import io.shiftleft.codepropertygraph.generated.Operators
+import io.shiftleft.semanticcpg.language._
+import io.shiftleft.semanticcpg.language.operatorextension.OpNodes
+import org.slf4j.{Logger, LoggerFactory}
+import overflowdb.BatchedUpdate
+
+import scala.io.Source
+
+/**
+ * Sets the return and parameter types for builtin functions with known function
+ * signatures.
+ *
+ * TODO: Need to handle variadic arguments.
+ */
+class PhpSetKnownTypesPass(cpg: Cpg) extends CpgPass(cpg) {
```

Comment:
This could likely be a bit simpler, and potentially be written as a `ForkJoinParallelCpgPass` and have the file be configurable, e.g.:

```scala
// Corresponds to a parsed row in the known functions file
case class KnownFunction(
     name: String,
     rTypes: Seq[String],               // return types
     pTypes: Seq[Seq[String]]      // Index 0 = parameter at P0
)

class PhpSetKnownTypesPass(cpg: Cpg, knownTypesFile: Option[File] = None) 
    extends ForkJoinParallelCpgPass[KnownFunction](cpg) {

    override def generateParts(): Array[KnownFunction] = {
     /* parse file and return each row as a KnownFunction object */
    }

   override def runOnPart(builder: overflowdb.BatchedUpdate.DiffGraphBuilder, part: KnownFunction): Unit = {
     /* calculate the result fo this part - this is done as a concurrent task */
   }
}
```